### ISSUE TITLE
Need sample for WinAppSDK / WinUI3 background tasks

### ISSUE THREAD

**Issue by jonwis** [microsoft/WindowsAppSDK-Samples#333]
https://learn.microsoft.com/windows/uwp/launch-resume/create-and-register-a-winmain-background-task shows how to register a background task for Win32 apps.  Please create a sample showing how to use this functionality from a WinAppSDK based app (with the AppInstance) and a WinUI3 app.

**Comment by godlytalias** [comment]
Samples are added in https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/BackgroundTask

**Comment by kinex** [comment]
Is it necessary to mix the app code with the background task code as in your sample (OutOfProc)? I would prefer having a separate project for the background task, but it is unclear how to implement it or whether it is even possible.

**Comment by godlytalias** [comment]
@kinex Would like to understand more about the 'mixing' part and what is the suggestion here, BackgroundTask registration happens on app during invoke of button [here](https://github.com/microsoft/WindowsAppSDK-Samples/blob/562c30912b387a77b1c1649dafae8fcfa43487b9/Samples/BackgroundTask/OutOfProc%20BackgroundTask/cpp-winui/BackgroundTaskBuilder/MainWindow.xaml.cpp#L70) and Background task implementations are done as a separate class [here](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Samples/BackgroundTask/OutOfProc%20BackgroundTask/cpp-winui/BackgroundTaskBuilder/BackgroundTask.cpp).
Are you saying to move the implementation to separate project instead of 'class' ?

**Comment by kinex** [comment]
@godlytalias Yes, I would like to have all background task related stuff in a separate project (including `Program.cs, ComServer.cs and BackgroundTask.cs` if possible). Only the background task registration would happen in the main WinUI 3 app project. I think that design would be much better for maintenance and extensibility.

Something like this: https://github.com/Osirisoo0O/WASDKSample/tree/master/src/BackgroundTask (the implementation is very different from your sample, but for some reason I can't even build that project).

I am porting UWP app to WinUI 3 and in my UWP app the background task is already a separate project.

**Comment by godlytalias** [comment]
@kinex `Program.cs` is the entry point of the application, `ComServer.cs` is a generic code of setting up COM server, it is not a logic that is specific to BackgroundTask. So we would be left with only `BackgroundTask.cs`, and that is why a separate project was not created just for that class.

**Comment by nbevans** [comment]
This sample project is super confusing. It appears to mix at least two different background task approaches (ComServer and the BGTask stuff)

Highly confusing. It's not clear which parts are needed to achieve the WinAppSDKBGTask approach.

The official documentation also mixes up between UWP, WinUI 3, WinAppSDK,.... It's all super bizarre and super confusing. Why is this so hard?

**Comment by nbevans** [comment]
Another issue is that the Program.cs expects a typical `static void Main(args)` entry point. But a MAUI application does not provide the arguments. So how will this work?

https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Samples/BackgroundTask/OutOfProc%20BackgroundTask/cs-winui/BackgroundTaskBuilder/Program.cs
